src/optimizer/filter_push_down_optimizer.cpp

```diff
@@ -120,8 +120,9 @@ std::shared_ptr<LogicalOperator> FilterPushDownOptimizer::visitCrossProductRepla
     if (joinConditions.empty()) { // Nothing to push down. Terminate.
         return finishPushDown(op);
     }
+    // we don't need cardinality after planning is complete
     auto hashJoin = std::make_shared<LogicalHashJoin>(joinConditions, JoinType::INNER,
-        nullptr /* mark */, op->getChild(0), op->getChild(1));
+        nullptr /* mark */, op->getChild(0), op->getChild(1), 0 /* cardinality */);
```

Comment:
Do we still need the cardinality here?

Reply:
Yes. we should carry the cardinality on. Or we should have another phase to re-propagate stats bottom-top. Otherwise we might lose some opportunity later for potential optimizations, also the printed plan won't have accurate cardinality, which is bad for plan debugging.

Reply:
updated comment to TODO